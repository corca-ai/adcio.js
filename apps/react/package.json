{
  "name": "demo-react",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "lint": "eslint .",
    "preview": "vite preview"
  },
  "dependencies": {
    "adcio.js": "workspace:*",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.22.1",
    "swiper": "^11.0.6"
  },
  "devDependencies": {
    "@types/react": "^18.2.55",
    "@types/react-dom": "^18.2.19",
    "@vitejs/plugin-react": "^4.2.1",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.5",
    "msw": "^2.0.10",
    "typescript": "^5.2.2",
    "vite-tsconfig-paths": "^4.3.1",
    "mock-api": "workspace:*"
  },
  "msw": {
    "workerDirectory": "public"
  }
}
