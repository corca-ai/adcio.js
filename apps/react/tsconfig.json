{
  "compilerOptions": {
    "allowJs": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "incremental": true,
    "types": ["node"],
    "jsx": "preserve",
    "baseUrl": ".",
    "skipLibCheck": true
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}
