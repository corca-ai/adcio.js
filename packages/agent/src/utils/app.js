/* eslint-disable */
import { devLog } from "./common";

export const sendProductRouterToApp = (path) => {
  try {
    devLog("Calling ProductRouter\nProductRouter params: ", path);
    if (typeof ProductRouter !== "undefined") {
      ProductRouter.postMessage(path); // for Android, Flutter
    } else if (window?.webkit?.messageHandlers?.ProductRouter) {
      window.webkit.messageHandlers.ProductRouter.postMessage(path); // for iOS
    }
    throw new Error(
      "There is no send method. Please get in touch with a Web developer.",
    );
  } catch (e) {
    devLog("--------ERROR--------");
    devLog("Failed to call ProductRouter with params: ", path);
    devLog("Your userAgent: ", window?.navigator?.userAgent);
    devLog("error: ", e);
    devLog("---------------------");
  }
};
