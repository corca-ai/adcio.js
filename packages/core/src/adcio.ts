import {
  AdcioConfig,
  AdcioCreateRecommendationBannersParams,
  AdcioCreateRecommendationBannersResponse,
  AdcioCreateRecommendationProductsParams,
  AdcioCreateRecommendationProductsResponse,
  AdcioFetchPlacementsParams,
  AdcioFetchPlacementsResponse,
  AdcioObserveImpressionParams,
  AdcioOnAddToCartParams,
  AdcioOnClickParams,
  AdcioOnImpressionParams,
  AdcioOnPageViewParams,
  AdcioOnPurchaseParams,
  AdcioParams,
} from "./adcio.interface";
import { AdcioAnalytics } from "./analytics/analytics";
import { AdcioCore } from "./core";
import { AdcioImpressionObserver } from "./impression-observer/impression-observer";
import { AdcioPlacement } from "./placement/placement";

export class Adcio {
  private readonly config: AdcioConfig;
  private adcioCore: AdcioCore;
  private adcioPlacement: AdcioPlacement;
  private adcioAnalytics: AdcioAnalytics;

  constructor(config: AdcioParams) {
    this.config = {
      apiEndpoint: "https://api.adcio.ai",
      receiverEndpoint: "https://receiver.adcio.ai",
      ...config,
    };
    this.adcioCore = new AdcioCore({
      clientId: this.config.clientId,
      customerId: this.config.customerId,
    });

    this.adcioPlacement = new AdcioPlacement({
      adcioCore: this.adcioCore,
      apiEndpoint: this.config.apiEndpoint,
    });
    this.adcioAnalytics = new AdcioAnalytics({
      adcioCore: this.adcioCore,
      receiverEndpoint: this.config.receiverEndpoint,
    });
  }

  public setCustomerId(customerId: string) {
    this.adcioCore.setCustomerId(customerId);
  }

  // AdcioAnalytics
  public onPageView(params: AdcioOnPageViewParams) {
    return this.adcioAnalytics.onPageView(params);
  }

  public onImpression(logOptions: AdcioOnImpressionParams) {
    return this.adcioAnalytics.onImpression(logOptions);
  }

  public onClick(logOptions: AdcioOnClickParams) {
    return this.adcioAnalytics.onClick(logOptions);
  }

  public onAddToCart(params: AdcioOnAddToCartParams) {
    return this.adcioAnalytics.onAddToCart(params);
  }

  public onPurchase(params: AdcioOnPurchaseParams) {
    return this.adcioAnalytics.onPurchase(params);
  }

  // AdcioImpressionDetector
  public observeImpression(params: AdcioObserveImpressionParams) {
    return new AdcioImpressionObserver({ filter: params.filter }).observe(
      params.element,
    );
  }

  // AdcioPlacement
  public async fetchPlacements(
    params: AdcioFetchPlacementsParams,
  ): Promise<AdcioFetchPlacementsResponse> {
    return this.adcioPlacement.fetchPlacements(params);
  }

  /**
   *  @deprecated
   */
  public async createSuggestionProducts(
    params: AdcioCreateRecommendationProductsParams,
  ): Promise<AdcioCreateRecommendationProductsResponse> {
    return this.adcioPlacement.createRecommendationProducts(params);
  }

  public async createRecommendationProducts(
    params: AdcioCreateRecommendationProductsParams,
  ): Promise<AdcioCreateRecommendationProductsResponse> {
    return this.adcioPlacement.createRecommendationProducts(params);
  }

  public async createRecommendationBanners(
    params: AdcioCreateRecommendationBannersParams,
  ): Promise<AdcioCreateRecommendationBannersResponse> {
    return this.adcioPlacement.createRecommendationBanners(params);
  }
}
