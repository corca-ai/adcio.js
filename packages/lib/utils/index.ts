export * from "./numbers";
export * from "./dom";
