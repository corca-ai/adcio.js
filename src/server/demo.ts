export const registeredPlacementIds = [
  "df7533ce-f504-41f5-b3e7-4aeb03a861f7",
  "0b6b7885-d952-4d21-86ed-f4d92154f9fc",
];

export const suggestionResponse = {
  placement: {
    id: "9f9f9b00-dc16-41c7-a5cd-f9a788d3d481",
    title: "For web test",
    type: "GRID",
    suggestionType: "RECOMMEND",
    displayCount: 3,
    activated: true,
    clientId: "30cb6fd0-17a5-4c56-b144-fef67de81bef",
    pageUrl: "https://docs.adcio.ai",
    injector: null,
    screenShot: "",
    property: "PERSONAL",
    createdAt: "2023-07-24T02:29:23.643Z",
    updatedAt: "2023-07-31T06:05:43.058Z",
    deletedAt: null,
  },
  suggestions: [
    {
      logOptions: {
        requestId: "2023081805292275799",
        adsetId: "8cc22880-f58a-43f0-b3de-a23d3ecf0bf5",
      },
      product: {
        id: "8cc22880-f58a-43f0-b3de-a23d3ecf0bf5",
        idOnStore: "302795144",
        storeId: "30cb6fd0-17a5-4c56-b144-fef67de81bef",
        sellerId: "30cb6fd0-17a5-4c56-b144-fef67de81bef",
        name: "키홀 페인트 레터링 티셔츠 화이트 SETS008WHITE",
        price: 38000,
        description:
          "The t-shirt is white with a blue-colored keyboard hole in the center. This design gives the t-shirt a modern and stylish look. It would suit best someone who enjoys trendy and unique clothing, and the unconventional keyboard hole adds a touch of personality to the garment. The t-shirt may evoke a feeling of creativity and individuality, as the wearer showcases their personal style and taste in clothing.",
        image: "https://product-image.wconcept.co.kr/productimg/image.jpg",
        includeInRecommendation: true,
        data: {
          tags: ["BEST"],
          color: "white",
          category: "short-sleeved t-shirt",
        },
        url: "https://adciomockapp.page.link/?link=https://adciomockapp.page.link/products?id%3D302795144&apn=ai.corca.webview_mock&isi=6451439461&ibi=ai.corca.webviewMock",
        deepLink:
          "https://adciomockapp.page.link/?link=https://adciomockapp.page.link/products?id%3D302795144&apn=ai.corca.webview_mock&isi=6451439461&ibi=ai.corca.webviewMock",
        caption: null,
        createdAt: "2023-05-30T14:24:20.263Z",
        updatedAt: "2023-05-30T14:24:20.263Z",
      },
    },
    {
      logOptions: {
        requestId: "2023081805292275799",
        adsetId: "9db03339-7e81-4e11-a130-39a3fab3a096",
      },
      product: {
        id: "9db03339-7e81-4e11-a130-39a3fab3a096",
        idOnStore: "302673758",
        storeId: "30cb6fd0-17a5-4c56-b144-fef67de81bef",
        sellerId: "30cb6fd0-17a5-4c56-b144-fef67de81bef",
        name: "AX자수 로고 패턴 후드 티셔츠A423131502-화이트",
        price: 202640,
        description:
          "This hoodie is predominantly white (with a small amount of grey), simple, and fashioned in a very bohemian style. It is described as being soft and comfortable. The hoodie can suit a hipster or a person seeking a casual and unconventional fashion statement. Overall, this hoodie evokes a feeling of relaxed, easygoing style that appeals to those who value individuality and nonconformity.",
        image:
          "https://product-image.wconcept.co.kr/productimg/image/img0/58/302673758_GG.jpg",
        includeInRecommendation: true,
        data: {
          tags: ["BEST"],
          color: "white",
          category: "hoodie",
        },
        url: "https://adciomockapp.page.link/?link=https://adciomockapp.page.link/products?id%3D302673758&apn=ai.corca.webview_mock&isi=6451439461&ibi=ai.corca.webviewMock",
        deepLink:
          "https://adciomockapp.page.link/?link=https://adciomockapp.page.link/products?id%3D302673758&apn=ai.corca.webview_mock&isi=6451439461&ibi=ai.corca.webviewMock",
        caption: null,
        createdAt: "2023-05-30T14:15:59.510Z",
        updatedAt: "2023-05-30T14:15:59.510Z",
      },
    },
    {
      logOptions: {
        requestId: "2023081805292275799",
        adsetId: "f2ed8893-f5a1-4ee2-941e-ebe7572ab8c2",
      },
      product: {
        id: "f2ed8893-f5a1-4ee2-941e-ebe7572ab8c2",
        idOnStore: "302529135",
        storeId: "30cb6fd0-17a5-4c56-b144-fef67de81bef",
        sellerId: "30cb6fd0-17a5-4c56-b144-fef67de81bef",
        name: "코튼 티셔츠 화이트 1150469002",
        price: 9900,
        description:
          "The color of this t-shirt is white. It has a very simple design with no printed or embroidered elements on it. The t-shirt appears to be a modern, trendy style suitable for a variety of individuals. The feeling it evokes is one of freshness, cleanliness, and minimalism. The overall impression of this t-shirt is that it is a versatile and stylish option for people who appreciate clean and contemporary clothing designs. The aesthetic it expresses is reminiscent of modernist clothing styles that emphasize simplicity and elegance.",
        image:
          "https://product-image.wconcept.co.kr/productimg/image/img0/35/302529135_LG.jpg",
        includeInRecommendation: true,
        data: {
          tags: ["BEST"],
          color: "white",
          category: "short-sleeved t-shirt",
        },
        url: "https://adciomockapp.page.link/?link=https://adciomockapp.page.link/products?id%3D302529135&apn=ai.corca.webview_mock&isi=6451439461&ibi=ai.corca.webviewMock",
        deepLink:
          "https://adciomockapp.page.link/?link=https://adciomockapp.page.link/products?id%3D302529135&apn=ai.corca.webview_mock&isi=6451439461&ibi=ai.corca.webviewMock",
        caption: null,
        createdAt: "2023-05-30T14:15:55.910Z",
        updatedAt: "2023-05-30T14:15:55.910Z",
      },
    },
  ],
};
