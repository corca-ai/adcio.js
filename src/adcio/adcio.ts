import { AdcioAnalytics } from "lib/analytics/analytics";
import { AdcioCore } from "lib/core";
import { FrontAPI } from "lib/front-api/front-api.interface";
import { AdcioImpressionObserver } from "lib/impression-observer/impression-observer";
import { AdcioPlacement } from "lib/placement/placement";
import {
  AdcioConfig,
  AdcioParams,
  AdcioCreateSuggestionParams,
  AdcioOnPageViewParams,
  AdcioOnAddToCartParams,
  AdcioOnClickParams,
  AdcioOnPurchaseParams,
  AdcioObserveImpressionParams,
  AdcioOnImpressionParams,
} from "./adcio.interface";

export class Adcio {
  private readonly config: AdcioConfig;
  private adcioCore: AdcioCore;
  private adcioPlacement: AdcioPlacement;
  private adcioAnalytics: AdcioAnalytics;

  constructor(config: AdcioParams) {
    this.config = {
      apiEndpoint: "https://api.adcio.ai",
      receiverEndpoint: "https://receiver.adcio.ai",
      ...config,
    };
    this.adcioCore = new AdcioCore({
      clientId: this.config.clientId,
      customerId: this.config.customerId,
    });

    this.adcioPlacement = new AdcioPlacement({
      adcioCore: this.adcioCore,
      apiEndpoint: this.config.apiEndpoint,
    });
    this.adcioAnalytics = new AdcioAnalytics({
      adcioCore: this.adcioCore,
      receiverEndpoint: this.config.receiverEndpoint,
    });
  }

  // AdcioAnalytics
  public onPageView(params: AdcioOnPageViewParams) {
    return this.adcioAnalytics.onPageView({
      path: window.location.pathname,
      title: document.title,
      referrer: document.referrer || undefined,
      ...params,
    });
  }

  public onImpression(logOptions: AdcioOnImpressionParams) {
    return this.adcioAnalytics.onImpression(logOptions);
  }

  public onClick(logOptions: AdcioOnClickParams) {
    return this.adcioAnalytics.onClick(logOptions);
  }

  public onAddToCart(params: AdcioOnAddToCartParams) {
    return this.adcioAnalytics.onAddToCart(params);
  }

  public onPurchase(params: AdcioOnPurchaseParams) {
    return this.adcioAnalytics.onPurchase(params);
  }

  // AdcioImpressionDetector
  public observeImpression(params: AdcioObserveImpressionParams) {
    return new AdcioImpressionObserver({ filter: params.filter }).observe(
      params.element,
    );
  }

  // AdcioPlacement
  public async createSuggestion(params: AdcioCreateSuggestionParams) {
    return this.adcioPlacement.createSuggestion(params);
  }

  public async log(frontApi: FrontAPI) {
    await frontApi.init();

    const promises: Promise<void>[] = [];

    const product = await frontApi.getProduct();
    promises.push(this.onPageView({ productIdOnStore: product?.idOnStore }));

    const carts = await frontApi.getCarts();
    if (carts && carts.length === 0) {
      // TODO: add to cart
      // const added = this.tracker.addedToCart(carts);
      // if (added.length > 0) {
      //   promises.push(
      //     ...added.map((cart) =>
      //       this.onAddToCart({
      //         cartId: cart.id,
      //         productIdOnStore: cart.productIdOnStore,
      //       }),
      //     ),
      //   );
      // }
    }

    const order = await frontApi.getOrder();
    if (order) {
      promises.push(
        ...order.products.map((product) =>
          this.onPurchase({
            orderId: order.id,
            amount: Number(
              product.subTotalPrice || product.price * product.quantity,
            ),
            quantity: product.quantity,
            productIdOnStore: product.idOnStore,
          }),
        ),
      );
    }
  }
}
