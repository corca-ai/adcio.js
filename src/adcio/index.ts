// Adcio
export { Adcio } from "./adcio";
export type {
  AdcioParams,
  AdcioConfig,
  AdcioOnPageViewParams,
  AdcioOnClickParams,
  AdcioOnAddToCartParams,
  AdcioOnPurchaseParams,
  AdcioObserveImpressionParams,
  AdcioCreateSuggestionParams,
} from "./adcio.interface";

// FrontAPI
export type { ClientAPI } from "lib/client-api";
export { Cafe24API } from "lib/client-api";

import { Cafe24API } from "lib/client-api";
export const frontApi = {
  cafe24: new Cafe24API(),
};

// utils
export {
  waitForDOM,
  waitForElement,
  getMeta,
  createElement,
  createNestedElement,
} from "lib/utils";
