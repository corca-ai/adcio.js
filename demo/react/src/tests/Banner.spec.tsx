import { act, render, screen, waitFor, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { UserEvent } from "@testing-library/user-event/dist/types/setup/setup";
import { suggestionResponse } from "mock-api/suggestion.demo";
import { Adcio } from "adcio.js";

import { wrapper } from "./utils/renderUI";
import BannerPage from "../pages/Banner";
import { afterEach, describe, vi, expect, it } from "vitest";

const setup = () => {
  console.log(Adcio);
  const adcioInstance = new Adcio({
    clientId: "your-client-id",
    customerId: "your-customer-id",
  });
  return render(<BannerPage adcioInstance={adcioInstance} />, {
    wrapper: (props) => wrapper(props, { route: "/" }),
  });
};

describe("Test SDK Banner", () => {
  it("test", async () => {
    const { container } = setup();
    await waitFor(() => {
      expect(screen.getByText("Loading...")).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(screen.getByText("[PC] Banner Test")).toBeInTheDocument();
    });
    expect(container).toMatchSnapshot();
  });
});
