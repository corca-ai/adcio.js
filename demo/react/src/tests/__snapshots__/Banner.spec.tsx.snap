// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test SDK Banner > test 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title"
    >
      [PC] Banner Test
    </div>
    <div
      class="swiper swiper-initialized swiper-horizontal mySwiper"
    >
      <div
        class="swiper-wrapper"
        style="transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://images.unsplash.com/photo-1705918519742-ff6bb5f1d034?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzgzNg&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              영화 '리프트'의 배우
김윤지 X 콜라보
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              넷플릭스 85개국 1위 영화 배우의
스타일리시한 애슬레저룩을 만나보세요.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://plus.unsplash.com/premium_photo-1697474429636-d5495b2e12a1?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzg2Mw&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              이번 시즌 대세는, 미니! NEW 반달백 출시
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              생활 방수도 되니까, 오염 걱정도 없어요.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://images.unsplash.com/photo-1706486003955-ed67dc1ed5bc?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzg3OQ&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              힙 부담 없는 인생 조거
이번 주말 등산에 입고 가요
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              하산 후에도 무릎 나옴도 걱정 없죠.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://images.unsplash.com/photo-1606041008023-472dfb5e530f?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTY4MTQ2NDIzNA&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              풋살부터 여행까지,
옷장 속 필수템 조거팬츠
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              두 가지 기장으로, 내 체형에 딱 맞게 입어요.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://images.unsplash.com/photo-1707757349249-c812bf8c600f?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzg5MQ&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              "바람은 막고 열기는 빼고"
일교차 심한 요즘 딱이죠
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              심지어 가벼우니까, 등산 · 조깅 만능이네요.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://images.unsplash.com/photo-1706816997326-3d0f825f8f88?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzkwNg&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              조깅부터 클라이밍까지,
"심지어 입고 벗기도 쉽죠"
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              고강도 운동엔 하이 서포트를 입어보세요.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://images.unsplash.com/photo-1705667834380-165981167727?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzkyMQ&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              "출근 준비 15분 절약해요"
사계절 링클프리 셔츠
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              골프 연습 갔다가 회사 가는 여유가 생기죠.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://plus.unsplash.com/premium_photo-1681223965358-af0cb4f398ec?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzkzMQ&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              허리가 쏙- 들어간 실루엣
"봄 라운딩만 기다려요"
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              니트카라 깃을 세우면 목탈 걱정도 없네요.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://images.unsplash.com/photo-1703231528215-739febd25d52?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzkzOQ&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              붙지 않는 면 티셔츠,
어깨가 넓어진 핏으로 만나요
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              기능성인지 아무도 모르니까, 색깔별로 쟁였어요.
            </p>
          </div>
        </div>
        <div
          class="swiper-slide"
        >
          <img
            alt=""
            src="https://images.unsplash.com/photo-1706126218054-a9c83cad2d03?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwxfDB8MXxyYW5kb218MHx8fHx8fHx8MTcwODQzNzk0OA&ixlib=rb-4.0.3&q=80&w=1080"
          />
          <div
            class="explain_banner"
          >
            <p
              class="title"
              style="color: rgba(255, 255, 255, 1);"
            >
              스타필드 수원
Grand Opening
            </p>
            <p
              class="subtitle"
              style="color: rgba(255, 255, 255, 1);"
            >
              감각적인 공간에서 보고 만지며 경험해 보세요.
            </p>
          </div>
        </div>
      </div>
      <div
        class="swiper-button-prev"
      />
      <div
        class="swiper-button-next"
      />
      <div
        class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets swiper-pagination-horizontal"
      />
    </div>
  </div>
</div>
`;
